version/version.go

```diff
@@ -64,6 +73,41 @@ func (v *Version) AtLeast(minVersion string) bool {
 	return v.Compare(minVersion) <= 0
 }
 
+// Check if a candidate is an upgrade/downgrade of the base version only in the specified position
+// Which corresponds to Major, Minor and Patch positions.
```

Comment:
"upgrade/downgrade" isn't really relevant as a description for this generic function.
I suggest replacing the abive two lines with - 
Compare the specified coordinate (major, minor or patch) between two versions.